### PR TITLE
Add TiDB limitations

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add TiDB limitations.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3166 https://github.com/pingcap/docs-cn/pull/3635
- Other reference link(s):

### CODE REVIEW

File: tidb-limitations.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+title: TiDB Limitations
+summary: Learn the limitations of TiDB.
```
Comment (lilin90) [inline tidb-limitations.md:3]:
```suggestion
summary: Learn the usage limitations of TiDB.
```

File: tidb-limitations.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+category: introduction
+---
+
+# TiDB Limitations
+
+This document describes the common limitations of TiDB, including the maximum identifier length and the maximum count of supported databases, tables, indexes, partitioned tables, and sequences.
```
Comment (lilin90) [inline tidb-limitations.md:9]:
```suggestion
This document describes the common usage limitations of TiDB, including the maximum identifier length and the maximum number of supported databases, tables, indexes, partitioned tables, and sequences.
```

File: tidb-limitations.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Limitation on identifier length
```
Comment (lilin90) [inline tidb-limitations.md:11]:
```suggestion
## Limitations on identifier length
```

File: tidb-limitations.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+| Identifier type | Maximum length (character) |
```
Comment (lilin90) [inline tidb-limitations.md:13]:
```suggestion
| Identifier type | Maximum length (number of characters allowed) |
```

File: tidb-limitations.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+|:---------|:--------------|
+| Database | 64 |
+| Table    | 64 |
+| Column   | 64 |
+| Index    | 64 |
+| View     | 64 |
+| Sequence | 64 |
+
+## Limitation on the total count of databases, tables, views, and connections
```
Comment (lilin90) [inline tidb-limitations.md:22]:
```suggestion
## Limitations on the total number of databases, tables, views, and connections
```

File: tidb-limitations.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+| Identifier type  | Maximum count   |
```
Comment (lilin90) [inline tidb-limitations.md:24]:
```suggestion
| Identifier type  | Maximum number  |
```

File: tidb-limitations.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+|:----------|:----------|
+| Databases | unlimited |
+| Tables    | unlimited |
+| Views     | unlimited |
+| Connections| unlimited|
+
+## Limitation on a single database
```
Comment (lilin90) [inline tidb-limitations.md:31]:
```suggestion
## Limitations on a single database
```

File: tidb-limitations.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+| Type       | Upper limit   |
+|:----------|:----------|
+| Tables    | unlimited  |
+
+## Limitation on a single table
```
Comment (lilin90) [inline tidb-limitations.md:37]:
```suggestion
## Limitations on a single table
```

File: tidb-limitations.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+|:----------|:----------|
+| Size       | 6 MB       |
+
+## Limitation on a single column
+
+| Type       | Upper limit   |
+|:----------|:----------|
+| Size       | 6 MB       |
+
+## Limitation on string types
```
Comment (lilin90) [inline tidb-limitations.md:59]: ditto

Comment (lilin90) [reply tidb-limitations.md:59]: In this context, the plural form of `limitation` is more commonly used. Please pay attention to this word.

File: tidb-limitations.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+| Type       | Upper limit   |
+|:----------|:----------|
+| CHAR       | 256 characters      |
+| BINARY     | 256 characters      |
+| VARBINARY  | 65535 characters    |
+| VARCHAR    | 16383 characters    |
+| TEXT       | 6MB bytes      |
```
Comment (lilin90) [inline tidb-limitations.md:67]:
```suggestion
| TEXT       | 6 MB bytes      |
```

File: tidb-limitations.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+| BLOB       | 6MB bytes      |
```
Comment (lilin90) [inline tidb-limitations.md:68]: ditto

File: tidb-limitations.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+## Limitation on SQL statements
```
Comment (lilin90) [inline tidb-limitations.md:70]: ditto

File: tidb-limitations.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+| Type       | Upper limit   |
+|:----------|:----------|
+| The number of SQL statements in a single transaction |  When the optimistic transaction is used and the transaction retry is enabled, the default upper limit is 5000, which can be changed through [`stmt-count-limit`](/tidb-configuration-file.md#stmt-count-limit). |
```
Comment (lilin90) [inline tidb-limitations.md:74]: Some meaning is missed. Try to be clearer in each column.
```suggestion
| The maximum number of SQL statements in a single transaction |  When the optimistic transaction is used and the transaction retry is enabled, the default upper limit is 5000, which can be modified using [`stmt-count-limit`](/tidb-configuration-file.md#stmt-count-limit). |
```

**Comment by TomShawn** [comment]
@lilin90 Comment addressed, PTAL again, thanks!

File: tidb-limitations.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+| Type       | Upper limit   |
+|:----------|:----------|
+| CHAR       | 256 characters      |
+| BINARY     | 256 characters      |
+| VARBINARY  | 65535 characters    |
+| VARCHAR    | 16383 characters    |
+| TEXT       | 6 MB bytes      |
+| BLOB       | 6 MB bytes      |
+
+## Limitation on SQL statements
```
Comment (lilin90) [inline tidb-limitations.md:70]:
```suggestion
## Limitations on SQL statements
```

**Pull request merged**